name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/AbstractPDFExtractor.java

```diff
@@ -110,7 +113,11 @@ public String getPDFAuthor()
 
             for (Item item : items)
             {
-                if (item.getSubject().getNote() == null || item.getSubject().getNote().trim().length() == 0)
+                Annotated subject = item.getSubject();
+
+                if (subject instanceof Transaction)
+                    ((Transaction) subject).setSource(filename);
+                else if (subject.getNote() == null || subject.getNote().trim().length() == 0)
```

Comment:
Der Dateiname "Quelle" sollte nicht mehr in dem Notizfeld gesetz werden.
Lediglich sollte trim(); begefügt werden und geprüft werden, ob getNote() != 0 ist.

`
if (subject.getNote() != null || subject.getNote().trim().length() != 0)
   item.getSubject().setNote(item.getSubject().getNote().trim());
`